### ISSUE TITLE
After `eslint --init`: Configuration for rule "valid-typeof" is invalid

### ISSUE THREAD

**Issue by KINFOO** [eslint/eslint#7219]
**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** 4.5.0
* **npm Version:** 2.15.9

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**
```js
module.exports = {
    "env": {
        "browser": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "accessor-pairs": "error",
        "array-bracket-spacing": "off",
        "array-callback-return": "off",
        "arrow-body-style": "error",
        "arrow-parens": "error",
        "arrow-spacing": "error",
        "block-scoped-var": "error",
        "block-spacing": "error",
        "brace-style": "off",
        "callback-return": "off",
        "camelcase": "off",
        "class-methods-use-this": "error",
        "comma-dangle": "off",
        "comma-spacing": "off",
        "comma-style": [
            "error",
            "last"
        ],
        "complexity": "off",
        "computed-property-spacing": [
            "error",
            "never"
        ],
        "consistent-return": "off",
        "consistent-this": "off",
        "curly": "off",
        "default-case": "off",
        "dot-location": [
            "error",
            "property"
        ],
        "dot-notation": "off",
        "eol-last": "off",
        "eqeqeq": "off",
        "func-call-spacing": "error",
        "func-names": "off",
        "func-style": "off",
        "generator-star-spacing": "error",
        "global-require": "off",
        "guard-for-in": "off",
        "handle-callback-err": "off",
        "id-blacklist": "error",
        "id-length": "off",
        "id-match": "error",
        "indent": "off",
        "init-declarations": "off",
        "jsx-quotes": "error",
        "key-spacing": "off",
        "keyword-spacing": "off",
        "linebreak-style": "off",
        "lines-around-comment": "off",
        "max-depth": "off",
        "max-len": "off",
        "max-lines": "off",
        "max-nested-callbacks": "error",
        "max-params": "off",
        "max-statements": "off",
        "max-statements-per-line": "error",
        "multiline-ternary": "off",
        "new-parens": "error",
        "newline-after-var": "off",
        "newline-before-return": "off",
        "newline-per-chained-call": "off",
        "no-alert": "error",
        "no-array-constructor": "error",
        "no-bitwise": "off",
        "no-caller": "error",
        "no-catch-shadow": "error",
        "no-cond-assign": [
            "error",
            "except-parens"
        ],
        "no-confusing-arrow": "error",
        "no-continue": "error",
        "no-div-regex": "error",
        "no-duplicate-imports": "error",
        "no-else-return": "off",
        "no-empty-function": "off",
        "no-eq-null": "error",
        "no-eval": "off",
        "no-extend-native": "off",
        "no-extra-bind": "error",
        "no-extra-label": "error",
        "no-extra-parens": "off",
        "no-floating-decimal": "error",
        "no-global-assign": "error",
        "no-implicit-globals": "error",
        "no-implied-eval": "error",
        "no-inline-comments": "off",
        "no-inner-declarations": [
            "error",
            "functions"
        ],
        "no-invalid-this": "off",
        "no-iterator": "error",
        "no-label-var": "error",
        "no-labels": "error",
        "no-lone-blocks": "error",
        "no-lonely-if": "off",
        "no-loop-func": "error",
        "no-magic-numbers": "off",
        "no-mixed-operators": "off",
        "no-mixed-requires": "off",
        "no-multi-spaces": "off",
        "no-multi-str": "error",
        "no-multiple-empty-lines": "off",
        "no-negated-condition": "off",
        "no-nested-ternary": "error",
        "no-new": "off",
        "no-new-func": "error",
        "no-new-object": "error",
        "no-new-require": "error",
        "no-new-wrappers": "error",
        "no-octal-escape": "error",
        "no-param-reassign": "off",
        "no-path-concat": "error",
        "no-plusplus": "off",
        "no-process-env": "off",
        "no-process-exit": "error",
        "no-proto": "error",
        "no-prototype-builtins": "error",
        "no-restricted-globals": "error",
        "no-restricted-imports": "error",
        "no-restricted-modules": "error",
        "no-restricted-syntax": "error",
        "no-return-assign": "off",
        "no-script-url": "off",
        "no-self-compare": "error",
        "no-sequences": "error",
        "no-shadow": "off",
        "no-shadow-restricted-names": "error",
        "no-spaced-func": "error",
        "no-sync": "off",
        "no-tabs": "off",
        "no-template-curly-in-string": "error",
        "no-ternary": "off",
        "no-throw-literal": "error",
        "no-trailing-spaces": "off",
        "no-undef-init": "error",
        "no-undefined": "off",
        "no-underscore-dangle": "off",
        "no-unmodified-loop-condition": "error",
        "no-unneeded-ternary": "off",
        "no-unsafe-negation": "error",
        "no-unused-expressions": "off",
        "no-use-before-define": "off",
        "no-useless-call": "off",
        "no-useless-computed-key": "error",
        "no-useless-concat": "off",
        "no-useless-constructor": "error",
        "no-useless-escape": "off",
        "no-useless-rename": "error",
        "no-var": "off",
        "no-void": "error",
        "no-warning-comments": "off",
        "no-whitespace-before-property": "error",
        "no-with": "error",
        "object-curly-newline": "off",
        "object-curly-spacing": "off",
        "object-property-newline": "error",
        "object-shorthand": "off",
        "one-var": "off",
        "one-var-declaration-per-line": "off",
        "operator-assignment": "off",
        "operator-linebreak": "off",
        "padded-blocks": "off",
        "prefer-arrow-callback": "off",
        "prefer-const": "error",
        "prefer-reflect": "off",
        "prefer-rest-params": "off",
        "prefer-spread": "off",
        "prefer-template": "off",
        "quote-props": "off",
        "quotes": "off",
        "radix": "off",
        "require-jsdoc": "off",
        "rest-spread-spacing": "error",
        "semi": "off",
        "semi-spacing": "off",
        "sort-imports": "error",
        "sort-keys": "off",
        "sort-vars": "off",
        "space-before-blocks": "off",
        "space-before-function-paren": "off",
        "space-in-parens": "off",
        "space-infix-ops": "off",
        "space-unary-ops": [
            "error",
            {
                "nonwords": false,
                "words": false
            }
        ],
        "spaced-comment": "off",
        "strict": "off",
        "symbol-description": "error",
        "template-curly-spacing": "error",
        "unicode-bom": [
            "error",
            "never"
        ],
        "valid-jsdoc": "off",
        "valid-typeof": [
            "error",
            {
                "requireStringLiterals": false
            }
        ],
        "vars-on-top": "off",
        "wrap-iife": "off",
        "wrap-regex": "error",
        "yield-star-spacing": "error",
        "yoda": [
            "error",
            "never"
        ]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
I ran `eslint --init`.
Then ran `Atom` with [linter-eslint](https://github.com/AtomLinter/linter-eslint) installed.

**What did you expect to happen?**
I expected [linter-eslint](https://github.com/AtomLinter/linter-eslint) to enable me to validate ES6 JavaScript.
More precisely, I expected `eslint --init` to produce a valid `~/.eslintrc.js`.
Current faulty section of `~/.eslintrc.js` is:
```js
        "valid-typeof": [
            "error",
            {
                "requireStringLiterals": false
            }
        ]
```
**What actually happened? Please include the actual, raw output from ESLint.**
Opening `Atom` I got this right away:
![atomerror](https://cloud.githubusercontent.com/assets/154818/18781184/0ef718ba-8181-11e6-9e47-3e995c41a3d5.png)
Full stack is:
```
Error: /home/kevin/.eslintrc.js: Configuration for rule "valid-typeof" is invalid: Value "[object Object]" has more items than allowed.
Error: /home/kevin/.eslintrc.js:
	Configuration for rule "valid-typeof" is invalid:
	Value "[object Object]" has more items than allowed.
    at validateRuleOptions (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-validator.js:116:15)
    at /home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-validator.js:163:13
    at Array.forEach (native)
    at Object.validate (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-validator.js:162:35)
    at Object.load (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:523:19)
    at loadConfig (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:64:33)
    at new Config (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:197:38)
    at CLIEngine.executeOnText (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:750:28)
    at Object.cli.execute (/home/kevin/.atom/packages/linter-eslint/node_modules/eslint/lib/cli.js:181:36)
    at lintJob (/home/kevin/.atom/packages/linter-eslint/lib/worker.js:37:10)
```

Thanks for your work

**Comment by ilyavolodin** [comment]
Most likely what is happening is your local version of ESLint is different from your global version of ESLint. You ran `--init` with global version, but `linter-eslint` uses local version. Try running the following:
```sh
eslint -v
node_modules/.bin/eslint -v
```
from the directory of your project and see if the versions match. If not, either rerun `--init` with the local version, or make sure both of them match.

**Comment by KINFOO** [comment]
Well,

After running this,
```sh
$ rm -fr $( find /w/av-ui -name node_modules -type d )
$ eslint --init                    
? How would you like to configure ESLint? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? /w/av-ui
? What format do you want your config file to be in? JavaScript
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? Yes
? Where will your code run? Browser, Node
? Do you use CommonJS? No
? Do you use JSX? No
Determining Config: 100% [==============================] 78.6s elapsed, eta 0.0s 


Enabled 127 out of 230 rules based on 1184 files.
Successfully created .eslintrc.js file in /home/kevin
```

There is no issue opening same `Atom`...

Thanks @ilyavolodin
